openfasoc/generators/glayout/glayout/flow/blocks/composite/regulated_cascoded_current_mirror/regulated_cascoded_current_mirror.py

```diff
@@ -0,0 +1,374 @@
+from typing import Optional, Union 
+
+def delete_files_in_directory(directory_path):
+   try:
+     files = listdir(directory_path)
+     for file in files:
+       file_path = path.join(directory_path, file)
+       if path.isfile(file_path):
+         try:
+            chmod(file_path, 0o777)
+            remove(file_path)
+         except PermissionError:
+               print(f"PermissionError: Operation not permitted for {file_path}")
+     print("All files deleted successfully.")
+   except OSError:
+     print("Error occurred while deleting files.")
+
+global PDK_ROOT
+if 'PDK_ROOT' in environ:
+	PDK_ROOT = str(Path(environ['PDK_ROOT']).resolve())
+else:
+	PDK_ROOT = "/usr/bin/miniconda3/share/pdk/"
+ 
+def generate_current_mirror_netlist(
+    pdk: MappedPDK,
+    instance_name: str,
+    CM_size: tuple[float, float, int, int],  # (width, length, multipliers, fingers)
+    drain_net_A: str,
+    gate_net: str,
+    source_net: str,
+    drain_net_B: str,
+    transistor_type: str = "nfet",
+    bulk_net: str = None,
+    proposed_ground: str = None,  # Proposed ground net
+    dummy: bool = True,
+    subckt_only: bool = False,
+) -> Netlist:
+    """Generate a netlist for a current mirror."""
+
+    if bulk_net is None:
+        bulk_net = "VDD" if transistor_type.lower() == "pfet" else "VSS"
+
+    width = CM_size[0]
+    length = CM_size[1]
+    multipliers = CM_size[2]  
+    mtop = multipliers if subckt_only else 1
+    #mtop = multipliers * 2 if dummy else multipliers # Double the multiplier to account for the dummies
+    fingers =  CM_size[3] # Number of fingers of the interdigitized fets
+
+
+    model_name = pdk.models[transistor_type.lower()]
+
+    circuit_name = instance_name
+    nodes = [drain_net_A, drain_net_B, proposed_ground]
+
+    source_netlist = f".subckt {circuit_name} {' '.join(nodes)}\n"
+
+    #source_netlist += f"V{proposed_ground}1 ({proposed_ground} {bulk_net}) 0\n" #Proposed ground connection
+
+
+    # Generating only two transistors (one on each side):
+    source_netlist += f"XA {drain_net_A} {gate_net} {source_net} {bulk_net} {model_name} l={length} w={width} m={mtop} nf={fingers}\n"
+    source_netlist += f"XB {drain_net_B} {gate_net} {source_net} {bulk_net} {model_name} l={length} w={width} m={mtop} nf={fingers}\n"
+    if dummy:
+        source_netlist += f"XDUMMY {bulk_net} {bulk_net} {bulk_net} {bulk_net} {model_name} l={length} w={width} m={mtop}\n"
+    source_netlist += ".ends " + circuit_name
+
+
+    instance_format = "X{name} {nodes} {circuit_name} l={length} w={width} m={mult} nf={fingers}"
+
+    return Netlist(
+        circuit_name=circuit_name,
+        nodes=nodes,
+        source_netlist=source_netlist,
+        instance_format=instance_format,
+        parameters={
+            "model": model_name,
+            "width": width,
+            "length": length,
+            'mult': multipliers,
+        },
+    )
+
+# @validate_arguments
+def CurrentMirror(
```

Comment:
Can this function name be changed to specify that it is a cascode current mirror?